icechunk-python/src/lib.rs

```diff
@@ -88,10 +88,61 @@ fn initialize_logs(py: Python) -> PyResult<()> {
     Ok(())
 }
 
+fn extract_module_names(filter: &str) -> Vec<&str> {
```

Comment:
this should return an iterator ... something like `impl Iterator<Item=&str>`, by skipping the `collect` call